client/src/containers/Gene/GeneActivitiesTable.js

```diff
@@ -78,49 +78,66 @@ class GeneActivitiesTable extends Component {
     );
   };
 
-  getActivityDescriptor = (activityItem, selfGeneId) => {
-    const entityKeys = [
-      'provenance/merged-from',
-      'provenance/merged-into',
-      'provenance/split-from',
-      'provenance/split-into',
-    ];
-    const selfEntityStub = {
-      'gene/id': selfGeneId,
-    };
-    const [entityKey] = entityKeys.filter(
-      (key) => activityItem[key] && activityItem[key]['gene/id'] === selfGeneId
-    );
-    const [relatedEntityKey] = entityKeys.filter(
-      (key) => activityItem[key] && activityItem[key]['gene/id'] !== selfGeneId
+  getActivityDescriptor = (activityItem = {}, selfGeneId) => {
+    const what = activityItem['provenance/what'];
+    const { statusChange, relatedGeneId } = (activityItem.changes || []).reduce(
+      (result, change) => {
+        const { attr, value, added } = change || {};
+        if (attr === 'gene/merges' && added) {
+          return {
+            ...result,
+            relatedGeneId: value,
+          };
+        } else if (attr === 'gene/splits' && added) {
+          return {
+            ...result,
+            relatedGeneId: value,
+          };
+        } else if (attr === 'gene/status' && added) {
+          return {
+            ...result,
+            statusChange: value,
+          };
+        } else {
+          return {
+            ...result,
+          };
+        }
+      },
+      {}
     );
 
     let eventType;
-    if (entityKey) {
-      if (activityItem['provenance/merged-into'] === activityItem[entityKey]) {
-        eventType = 'merge_from';
-      } else if (
-        activityItem['provenance/merged-from'] === activityItem[entityKey]
-      ) {
-        eventType = 'merge_into';
-      } else if (
-        activityItem['provenance/split-into'] === activityItem[entityKey]
-      ) {
-        eventType = 'split_from';
-      } else if (
-        activityItem['provenance/split-from'] === activityItem[entityKey]
-      ) {
-        eventType = 'split_into';
-      }
+    if (what === 'event/merge-genes' && !statusChange) {
+      eventType = 'merge_from';
+    } else if (
+      what === 'event/merge-genes' &&
+      statusChange === 'gene.status/dead'
+    ) {
+      eventType = 'merge_into';
```

Comment:
```suggestion
      eventType = 'merged_into';
```

Reply:
I have never seen inline suggestion in action before. It's really nice.

I would prefer to leave "merge_into" and "merge_from" in present tense, because: 
- it's more consistent with values returned by provenance/what 
- (more of a personal taste) I'm okay with reading log entries in present tense, as I interpret it as being written when things are happening. But I don't know if others might find it confusing.

That being said, if past tense is preferred here, we would need a mapping (to past tense) for all the values returned by provenance/what.

Reply:
@sibyl229  hehe, I prefer past tense (because it has happened) - was/is certainly the intention.

`provenance` should only contain past/neutral-tense keys (full list):
 `when`
 `what`
 `who`
 `why` 

`grep provenance resources/schema/*edn`

n.b provenance keys merged-into merged-from, split-from, split-into have been deprecated (not used by the app).

Reply:
@sibyl229 expect my JS suggestions contained invalid syntax :weary: